ui/app/utils/validators.js

```diff
@@ -3,17 +3,20 @@ import { isPresent } from '@ember/utils';
 export const presence = (value) => isPresent(value);
 
 export const length = (value, { nullable = false, min, max } = {}) => {
-  let isValid = nullable;
-  if (typeof value === 'string') {
-    const underMin = min && value.length < min;
-    const overMax = max && value.length > max;
-    isValid = underMin || overMax ? false : true;
+  if (!min && !max) return;
+  // value could be an integer if the attr has a default value of some number
+  const valueLength = value?.toString().length;
+  if (valueLength) {
+    const underMin = min && valueLength < min;
+    const overMax = max && valueLength > max;
+    return underMin || overMax ? false : true;
   }
-  return isValid;
+  return nullable;
 };
 
 export const number = (value, { nullable = false, asString } = {}) => {
-  if (!value) return nullable;
+  // since 0 is falsy, !value is true even though 0 is valid here
```

Comment:
This was source part I of the bug. All validations are treated as a string, even tough default integer values only become a string once the user has interacted with the form. So in KV for example, the validation decorator attempts to validate the default input of `0` and since`!0 = true` the check is invalid.

Alternative approach - since they are form values, do we want all of our default inputs to be strings? (This would also address the issue above)